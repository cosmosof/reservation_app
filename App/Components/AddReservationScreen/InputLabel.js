// @flow

import * as React from 'react';
import {Text} from 'react-native';

import styles from './Styles/InputLabelStyles';

  type Props = {
    containerStyle?: Object,
    label?: string,
    styles: Object,
  };

export default function InputLabel(props: Props) {
  const {containerStyle, label} = props;
  return (
    <Text
      style={{
        ...styles.container,
        ...containerStyle,
      }}
    >
      {label}
    </Text>
  );
}

InputLabel.defaultProps = {
  containerStyle: undefined,
  label: 'add a label',
};
