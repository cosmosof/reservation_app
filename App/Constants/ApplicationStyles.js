import {Platform} from 'react-native';
import Fonts from './Fonts';
import Colors from './Colors';

const ApplicationStyles = {
  navbarHeader: {
    color: Colors.darkMatBlue2,
    fontSize: 18,
    fontWeight: '600',
    fontFamily: Platform.OS == 'ios' ? Fonts.typeIOS.condensed : Fonts.typeAndroid.condensed,
  },
};

export default {
  ApplicationStyles,
};
