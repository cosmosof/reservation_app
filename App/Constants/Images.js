const images = {
  logo: require('../Images/Relate.png'),
};

export default images;
