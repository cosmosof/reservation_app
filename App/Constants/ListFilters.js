export default [
  {
    index: 0,
    name: 'guest name',
    value: 'name',
  },
  {
    index: 1,
    name: 'in guest names',
    value: 'name_contains',
  },
  {
    index: 3,
    name: 'name starts with',
    value: 'name_starts_with',
  },
  {
    index: 4,
    name: 'hotel name',
    value: 'hotelName',
  },
  {
    index: 5,
    name: 'in hotel names',
    value: 'hotelName_contains',
  },
  {
    index: 7,
    name: 'hotel names start with',
    value: 'hotelName_starts_with',
  },
  {
    index: 9,
    name: 'arrival date',
    value: 'arrivalDate',
  },
  {
    index: 10,
    name: 'departure date',
    value: 'departureDate',
  },
];
