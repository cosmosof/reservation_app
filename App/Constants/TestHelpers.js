const noop = () => {};
const sampleText = 'Sample text';
export default {
  noop,
  sampleText,
};
