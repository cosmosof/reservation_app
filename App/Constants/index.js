import Colors from './Colors';
import Fonts from './Fonts';
import Metrics from './Metrics';
import Images from './Images';
import ListFilters from './ListFilters';
import ApplicationStyles from './ApplicationStyles';
import TestHelpers from './TestHelpers';

export {Colors, Fonts, Images, Metrics, ListFilters, ApplicationStyles, TestHelpers};
