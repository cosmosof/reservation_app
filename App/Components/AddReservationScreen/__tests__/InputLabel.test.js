import React from 'react';
import {shallow} from 'enzyme';
import InputLabel from '../InputLabel';
import {TestHelpers} from '../../../Constants';

const {sampleText} = TestHelpers;

describe('<InputLabel>', () => {
  const Component = <InputLabel />;
  describe('Structure', () => {
    it('renders correctly', () => {
      const wrapper = shallow(Component);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if containerStyle prop is set', () => {
      const Clone = React.cloneElement(Component, {containerStyle: {color: 'red'}});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if title prop is set', () => {
      const Clone = React.cloneElement(Component, {label: sampleText});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
