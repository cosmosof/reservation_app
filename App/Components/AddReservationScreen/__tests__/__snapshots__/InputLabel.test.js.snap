// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputLabel> Structure renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#00A8E8",
      "fontFamily": "AvenirNextCondensed-DemiBold",
      "fontSize": 14,
      "fontWeight": "500",
      "paddingBottom": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 90,
    }
  }
>
  add a label
</Text>
`;

exports[`<InputLabel> Structure renders correctly if containerStyle prop is set 1`] = `
<Text
  style={
    Object {
      "color": "red",
      "fontFamily": "AvenirNextCondensed-DemiBold",
      "fontSize": 14,
      "fontWeight": "500",
      "paddingBottom": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 90,
    }
  }
>
  add a label
</Text>
`;

exports[`<InputLabel> Structure renders correctly if title prop is set 1`] = `
<Text
  style={
    Object {
      "color": "#00A8E8",
      "fontFamily": "AvenirNextCondensed-DemiBold",
      "fontSize": 14,
      "fontWeight": "500",
      "paddingBottom": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 90,
    }
  }
>
  Sample text
</Text>
`;
